code/datums/ai/_ai_controller.dm

```diff
@@ -432,9 +441,22 @@ multiple modular subtrees with behaviors
  */
 /datum/ai_controller/proc/set_blackboard_key_assoc_lazylist(key, thing, value)
 	LAZYINITLIST(blackboard[key])
+	// Don't do anything if it's already got this value
+	if (blackboard[key][thing] == value)
+		return
+
 	TRACK_AI_DATUM_TARGET(thing, key)
 	TRACK_AI_DATUM_TARGET(value, key)
 	blackboard[key][thing] = value
+	post_blackboard_key_set(key)
+
+/**
+ * Called after we set a blackboard key, forwards signal information.
+ */
+/datum/ai_controller/proc/post_blackboard_key_set(key)
+	if (!pawn)
```

Comment:
```suggestion
	if (isnull(pawn))
```

why'd you name your commit "i guess it can be null there" and not use `isnull()`